tests/integration/targets/aci_cloud_aws_provider/tasks/main.yml

```diff
@@ -0,0 +1,138 @@
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("info") }}'
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for aws cloud site
+  when: 
+  - query_cloud.current | length > 0
+  - query_cloud.current.0.cloudProvP.attributes.vendor == "aws"  # This condition will execute only on cloud aws
+  block:  # block specifies execution of tasks within, based on conditions
+  
+  - name: Ensure tenant doesn't exist
+    cisco.aci.aci_tenant:
+      <<: *aci_info
+      tenant: ansible_test
+      state: absent
+
+  - name: Create tenant
+    cisco.aci.aci_tenant:
+      <<: *aci_info
+      tenant: ansible_test
+      state: present
+      
+  - name: Create aws provider
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: yes
+      state: present
+    register: aws_present
+
+  - name: Create aws provider again
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: yes
+      state: present
+    register: aws_present_again
+
+  - name: Change to untrusted aws provider 
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: no
+      state: present
+    register: aws_present_not_trusted
+    ignore_errors: true
+
+  - name: Delete aws provider 
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: yes
+      state: absent
+    register: aws_absent
+
+  - name: Create aws provider again after deletion as not trusted
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: no
+      access_key_id: 1
+      secret_access_key: 1
+      email: aws@gmail.com
+      http_proxy: "on"
+      state: present
+    register: aws_present_new
+
+  - name: Create aws provider again after deletion as not trusted
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      account_id: 111111111111
+      is_trusted: no
+      access_key_id: 1
+      secret_access_key: 1
+      email: aws@gmail.com
+      http_proxy: "on"
+      is_account_in_org: yes
+      state: present
+    register: aws_present_add_in_org
+
+  - name: Query aws provider
+    cisco.aci.aci_cloud_aws_provider:
+      <<: *aci_info
+      tenant: ansible_test
+      state: query
+    register: aws_query
+  
```

Comment:
Add Query all test

Reply:
That would display all the tenants including the ones that are not associated with AWS. Is that desirable?